crates/evm/src/evm/system_contracts/test/WitnessUtils.t.sol

```diff
@@ -0,0 +1,162 @@
+    bytes4 locktime = hex"00000000";
+    bytes32 wtxId = hex"ff63ed0cc02c85a7aef6510625b6897abe1b969293f5db2991804e23f2562df1";
+    bytes wExample = hex"06096af9822fdb33bbfb940c6e6ea3d182c36de9c428de5606dbb8f257b63a10b66f7e408fcf8ed73ce3c3abf2aa28d809ef64e1bc35146424d00e072631fe3a08749c3380609e8fcd080582e984c1b5026351016404031f73220a964e3d835c33a0be6ca70206c9044dbc1b2c04b3411e2a";
+    bytes badWitness1 = hex"06096af9822fdb33bbfb940c6e6ea3d182c36de9c428de5606dbb8f257b63a10b66f7e408fcf8ed73ce3c3abf2aa28d809ef64e1bc35146424d00e072631fe3a08749c3380609e8fcd080582e984c1b5026351016404031f73220a964e3d835c33a0be6ca70206c9044dbc1b2c04b3411e";
+    bytes badWitness2 = hex"06096af9822fdb33bbfb940c6e6ea3d182c36de9c428de5606dbb8f257b63a10b66f7e408fcf8ed73ce3c3abf2aa28d809ef64e1bc35146424d00e072631fe3a08749c3380609e8fcd080582e984c1b5026351016404031f73220a964e3d835c33a0be6ca70206c9044dbc1b2c04b3411e2a32";
+    bytes badWitness3 = hex"06096af9822fdb33bbfb940c6e6ea3d182c36de9c428de5606dbb8f257b63a10b66f7e408fcf8ed73ce3c3abf2aa28d809ef64e1bc35146424d00e072631fe3a08749c3380609e8fcd080582e984c1b5026351016404031f73220a964e3d835c33a0be6ca70206c9044dbc1b2c04b3411e";
+    bytes badArgWitness = hex"0101ff";
+    bytes badArgWitness2 = hex"0101ff01ff";
+    // Helper function to run Python script and get its output
+    function getRandomWitness() public returns (bytes memory) {
+        string[] memory inputs = new string[](2);
+        inputs[0] = "python3";
+        inputs[1] = "test/randomWitnessGenerator.py";
+        
+        bytes memory result = vm.ffi(inputs);
+        return result;
+    }
+
+    function getRandomStackItem() public returns (bytes memory) {
+        string[] memory inputs = new string[](2);
+        inputs[0] = "python3";
+        inputs[1] = "test/randomStackItemGenerator.py";
```

Comment:
Is it really a proper way of gen random hex strings?

Reply:
I could not find a native random library in foundry to use for complex statements like this (fuzzing was not enough, it is generally deterministic and gives the same randomness in each iteration). This approach was also used by a lot of people, that's why I choose to do like this.